### PR TITLE
added tests for complex dtype support

### LINKED ISSUE #48519: torch.diag not implemented for 'ComplexDouble' dtype
## 🐛 Bug

[torch.diag](https://pytorch.org/docs/master/generated/torch.diag.html#torch.diag) is not implemented for ComplexDouble data type.
## To Reproduce

1. importing torch
2. Create a tensor of Complex Double type
3. Call `torch.diag` on the above created complex double dtype tensor
```
import torch
>>> x1 = torch.ones(1, dtype=torch.cdouble, requires_grad=False) #Creates a tensor of Complex Double type
>>> y = torch.diag(x1)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "diag" not implemented for 'ComplexDouble'
```
## Expected behavior
To return the diagonal of the given input complex double dtype tensor 

```
>>> x1 = torch.ones(1, dtype=torch.cdouble, requires_grad=False) #Create a tensor of Complex Double type
>>> y = torch.diag(x1)
#output
tensor([[1.+0.j]], dtype=torch.complex128)
```
## Environment

```
Collecting environment information...
PyTorch version: 1.7.0+cu110
Is debug build: True
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (Ubuntu 8.4.0-3ubuntu2) 8.4.0
Clang version: Could not collect
CMake version: version 3.18.2

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: 
GPU 0: GeForce GTX 1050 Ti

cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.17.4
[conda] blas                      1.0                         mkl  
[conda] magma-cuda110             2.5.2                         1    pytorch
[conda] mkl                       2020.2                      256  
[conda] mkl-include               2020.2                      256  
[conda] mkl-service               2.3.0            py38he904b0f_0  
[conda] mkl_fft                   1.2.0            py38h23d657b_0  
[conda] mkl_random                1.1.1            py38h0573a6f_0  
[conda] numpy                     1.19.2           py38h54aff64_0  
[conda] numpy-base                1.19.2           py38hfa32c7d_0  
[conda] torch                     1.8.0a0                   dev_0    <develop>
[conda] torchaudio                0.7.0                    pypi_0    pypi
[conda] torchvision               0.8.1+cu110              pypi_0    pypi

```

cc @ezyang @anjali411 @dylanbespalko @mruberry

### PR DESCRIPTION
**Pull request by Rajathbharadwaj**
Fixes #48519 
It looked like complex dtype supported was already added for `[torch.diagflat]`(https://github.com/pytorch/pytorch/issues/47499) and it looks like `torch.diag `uses `torch.diagflat` , so I've added the tests for the same.

### CODE REVIEW

File: test/test_tensor_creation_ops.py [hunk 1]
```diff
@@ -339,6 +339,11 @@ def test_diagflat(self, device):
         expected = torch.eye(4, dtype=torch.complex128)
         self.assertEqual(result, expected)
 
+        # Complex support for torch.diag
```
Comment (anjali411) [inline test/test_tensor_creation_ops.py:342]: let's add test for `torch.diag` in a separate method

Comment (Rajathbharadwaj) [reply test/test_tensor_creation_ops.py:342]: Okay.
Any pointers which method?

Comment (anjali411) [reply test/test_tensor_creation_ops.py:342]: I think it's ok to add a new test method in this file `test_diag`

Comment (Rajathbharadwaj) [reply test/test_tensor_creation_ops.py:342]: Alright,  on it.

Comment (Rajathbharadwaj) [reply test/test_tensor_creation_ops.py:342]: Hey, @anjali411, I'm unable find the `test_diag`  file, is it the `test_shape_ops` ?

**Comment by Rajathbharadwaj** [comment]
Hey @anjali411, any suggestions?

File: test/test_tensor_creation_ops.py (hunk 1 above)
Comment (anjali411) [reply test/test_tensor_creation_ops.py:342]: Hey @Rajathbharadwaj you can just create a new method in the same file:
```
test_diag_complex(self, device):
     result = torch.diag(torch.ones(4, dtype=torch.complex128))
     expected = torch.eye(4, dtype=torch.complex128)
     self.assertEqual(result, expected)
```

**Comment by Rajathbharadwaj** [comment]
Hey @anjali411 , I've added `test_diag_complex()` method in the same `test_tensor_creation_ops.py `

**Comment by anjali411** [comment]
Closing this since it looks like the issue was resolved in another PR

**Pull request closed without merging**